src/psyclone/generator.py

```diff
@@ -445,6 +447,17 @@ def main(args):
         print(str(err), file=sys.stderr)
         sys.exit(1)
 
+    try:
+        ModuleManager.get().add_search_path(args.directory)
+    except IOError:
+        # TODO #2011: We ignore the error here (indicating an invalid
+        # search path), since atm the kernel search path (args.directory)
+        # is checked later, and this way no existing code is broken (raising
```

Comment:
Could we therefore do this call later, *after* it has been checked?

Reply:
Yes, I've moved it. I kind of like it better to be done early, since the flow is easier, but ... I hope with #2011 that might all be cleaned up :)